File: 1950-sign-of-the-product-of-an-array/sign-of-the-product-of-an-array.py
class Solution:
    def arraySign(self, nums: List[int]) -> int:
        def signFunc(x):
            if x == 0:
                return 0
            elif x > 0:
                return 1
            else:
                return -1
        prod = 1
        for i in nums:
            if i == 0:
                prod = 0
                break
            prod *= i
        return signFunc(prod)
        

File: 1950-sign-of-the-product-of-an-array/sign-of-the-product-of-an-array.java
class Solution {
    public int arraySign(int[] nums) {
        int prod = 1;
        for (int i : nums){
            if (i == 0) prod = 0;
            else if (i > 0) prod *= 1;
            else prod *= -1;
        }
        return signFunc(prod);
    }

    public int signFunc(int x){
        if (x > 0) return 1;
        else if (x < 0) return -1;
        else return 0; 
    }
}